File: CS/DialogsDemo.Wpf/ViewModels/ThemedWindowModule/HeaderModels/BaseHeaderItemModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.Mvvm.POCO;

namespace DialogsDemo {
    public abstract class BaseHeaderItemModel {
        public virtual bool IsVisible { get; set; }
        public ThemedWindowViewModel BaseModel { get { return this.GetParentViewModel<ThemedWindowViewModel>(); } }

        public string ResourceKey { get; set; }
    }
}

File: VB/DialogsDemo.Wpf/ViewModels/ThemedWindowModule/HeaderModels/BaseHeaderItemModel.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.Mvvm.POCO

Namespace DialogsDemo
    Public MustInherit Class BaseHeaderItemModel
        Public Overridable Property IsVisible() As Boolean
        Public ReadOnly Property BaseModel() As ThemedWindowViewModel
            Get
                Return Me.GetParentViewModel(Of ThemedWindowViewModel)()
            End Get
        End Property

        Public Property ResourceKey() As String
    End Class
End Namespace
